src/disp_s1/solid_earth_tides.py

```diff
@@ -0,0 +1,247 @@
+
+def interpolate_set_components(
+    n: np.ndarray,
+    e: np.ndarray,
+    y_arr: np.ndarray,
+    x_arr: np.ndarray,
+    set_east: np.ndarray,
+    set_north: np.ndarray,
+    set_up: np.ndarray,
+) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+    """Interpolate SET components to the unwrapped interferogram grid."""
+    points = np.stack((n, e), axis=-1)
+    return (
+        RegularGridInterpolator((y_arr, x_arr), set_east)(points),
+        RegularGridInterpolator((y_arr, x_arr), set_north)(points),
+        RegularGridInterpolator((y_arr, x_arr), set_up)(points),
+    )
+
+
+def run_solid_grid(az_time, lat, lon, res_lat, res_lon):
+    """Run solid_grid from pysolid."""
+    az_time = pd.to_datetime(az_time)
+    return solid_grid(
+        az_time.year,
+        az_time.month,
+        az_time.day,
+        az_time.hour,
+        az_time.minute,
+        az_time.second,
+        lat,
+        res_lat,
+        1,
+        lon,
+        res_lon,
+        1,
+    )
+
+
+def calculate_time_ranges(
+    reference_start_time: pd.Timestamp,
+    reference_stop_time: pd.Timestamp,
+    secondary_start_time: pd.Timestamp,
+    secondary_stop_time: pd.Timestamp,
+    atr: dict,
+) -> Tuple[np.ndarray, np.ndarray]:
+    """Generate time ranges for reference and secondary files."""
+    ref_time_range = pd.to_datetime(
+        np.linspace(
+            reference_start_time.value, reference_stop_time.value, atr["LENGTH"]
+        )
+    )
+    sec_time_range = pd.to_datetime(
+        np.linspace(
+            secondary_start_time.value, secondary_stop_time.value, atr["LENGTH"]
+        )
+    )
+    ref_time_tile = np.tile(ref_time_range, (atr["WIDTH"], 1)).T
+    sec_time_tile = np.tile(sec_time_range, (atr["WIDTH"], 1)).T
+    return ref_time_tile, sec_time_tile
+
+
+def calculate_solid_earth_tides_correction(
+    ifgram_filename: Filename,
+    reference_cslc_file: Filename,
+    secondary_cslc_file: Filename,
+    los_east_file: Filename,
+    los_north_file: Filename,
+    reference_point: Optional[ReferencePoint] = None,
+) -> np.ndarray:
+    """Calculate the relative displacement correction for solid earth tides."""
+    # Load bounds, transform, CRS, and shape from the unwrapped interferogram
+    bounds, crs, affine_transform, (height, width) = load_raster_bounds_and_transform(
+        ifgram_filename
+    )
+    bounds_geo = transform_bounds_to_epsg4326(bounds, crs)
+    atr = generate_atr(bounds_geo, height, width)
+
+    # Extract timing information from the CSLC files
+    reference_start_time = pd.to_datetime(
+        get_zero_doppler_time(reference_cslc_file, type_="start")
+    )
+    reference_stop_time = pd.to_datetime(
+        get_zero_doppler_time(reference_cslc_file, type_="end")
+    )
+    secondary_start_time = pd.to_datetime(
+        get_zero_doppler_time(secondary_cslc_file, type_="start")
+    )
+    secondary_stop_time = pd.to_datetime(
+        get_zero_doppler_time(secondary_cslc_file, type_="end")
+    )
```

Comment:
Something I'm realizing upon second read-
We're passing one `reference_cslc_file`, and one for the secondary. These will be burst CSLCs, meaning just 1/9 the distance along track for the whole disp frame. This means we're not getting the right time bounds unless we pass the first CSLC to get the start time, and the last cslc to get the stop time. 

This is not just a problem here actually- you made me realized that I [messed this up a few PR's ago](https://github.com/opera-adt/disp-s1/commit/07a90632d6a7a04cf9a1778c885389ca0081971e#diff-9dd3a77a3705883ba9f899d7dc11030326ff84e1c4d785c41e0ee3f35b9fdd30R132-R134), and need to change it back to get the times for all CSLCs and take the min/max